### DISCUSSION TITLE
How can i add access the model inside the table column?

### DISCUSSION THREAD

**Discussion by maherelgamil** [filamentphp/filament#1458] [Help]
I need to present  a custom and do some actions inside `->default()` or `->formatStateUsing()` and need to catch the model some thing like `$this->resource->id`, or any other attribute

**Comment by intrepidws** [comment]
@maherelgamil Any chance you figured this out? I have this same need, specifically with the `formatStateUsing()` method.

**Reply by danharrin** [reply]
It's the `$record` property of the Closure, it should be documented, is that not the case?

Also, make sure you're not abusing `formatStateUsing()` when maybe you need `getStateUsing()`. One formats state that has already been retrieved, the other is used to get that state in the first place. Both work the same, but only the first has access to the pre-retrieved `$state`.

**Reply by intrepidws** [reply]
Thanks Dan. I read through the docs several times and didn't notice it, but see it now. I tend to look at the code examples rather than the other text, so that's probably why I missed it. Thanks for your help here.

**Comment by intrepidws** [comment]
This is what I ended up using:

```php
TextColumn::make('my_field')
    ->getStateUsing(function (Transformer $record) {
        // My logic using $record->id
    }),
```
